### ISSUE TITLE
Remove ioutil usage and leverage other Go1.16 features 

### ISSUE THREAD

**Issue by azdagron** [spiffe/spire#2222]
This is a small issue to track changes we should make to the codebase after #2210 is merged.

Some of these are required moving forward. Others are just nice-to-have.

- [x] Replace `io/ioutil` usage in favor of new functions added to the io and os packages. `io/ioutil` is deprecated.
- [x] Use `go install` in the Makefile to install dependencies now that it doesn't modify `go.mod` instead of the workaround building in a temporary directory that we're currently using
- [x]  Replace our hand-rolled "context cancellation on signal" code with the new `signal.NotifyContext`. This includes, but isn't necessarily limited to pkg/common/util.SignalListener.

**Comment by azdagron** [comment]
Leveraging `go install` was fixed with #2538.

**Comment by ryysud** [comment]
@azdagron Is this issue already fixed?

**Comment by azdagron** [comment]
It is now! Thanks!
